### PR TITLE
VPBlock: Support edit title and poster from upload (V6)

### PR DESCRIPTION
**Pull request by renatoagds**
Provide the experience to edit the title and poster at the upload moment.

It's mostly based from #24556 

#### Changes proposed in this Pull Request:

- Introduce new hooks to handle meta and poster upload/get.
- Split out `UploadProgress` and `UploadError` at own files.
- Include support at `/meta` endpoint to edit `title`.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

- Start a new post.
- Insert `VideoPress (beta)` block.
- Try to change the title and poster (from the library or set a frame).

#### Demo

_Selecting poster from Frame_

https://user-images.githubusercontent.com/1663717/185452581-0c300826-971c-4d8f-b55d-6c6e00488c32.mov

_Selecting poster from Library_

https://user-images.githubusercontent.com/1663717/185453116-4e221e98-f83d-48c3-93b7-2c9cbbc28731.mp4

### CODE REVIEW

**Review by retrofox — APPROVED** [review]
Tested. LGTM. 🎉

**Review by retrofox — CHANGES_REQUESTED** [review]
Sorry, after testing the PR I found a bug where the video is rendered after saving and hard-fresh.

image

It shows the media placeholder even when the block has a valid video source.

**Comment by renatoagds** [comment]
@retrofox thanks for checking it, I'll take a look

**Comment by renatoagds** [comment]
@retrofox fixed it, and looks like as side effect it fixed 1202184803119163-as-1202720808407728/f

File: projects/plugins/jetpack/extensions/blocks/videopress/v6/edit.js [hunk 1]
```diff
@@ -218,7 +218,7 @@ export default function VideoPressEdit( { attributes, setAttributes, isSelected,
 		} ),
 	} );
 
-	const [ isUploadingFile, setIsUploadingFile ] = useState( true );
+	const [ isUploadingFile, setIsUploadingFile ] = useState( ! attributes?.guid );
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/videopress/v6/edit.js:221]: nice one

**Review by retrofox — APPROVED** [review]
LGTM 🚀

**Comment by renatoagds** [comment]
r251064-wpcom

**Pull request merged**